website/upgrade-guides/0-11.html.markdown

```diff
@@ -93,8 +93,8 @@ situations:
   the descendent and the descendent no longer inherits attributes from
   its ancestor. Instead, each configuration is entirely distinct.
 
-* Only unaliased provider configurations can be automatically inherited from
-  ancestor modules. Aliased providers must be passed explicitly using
+* Only implicit provider configurations can be automatically inherited from
```

Comment:
The doc section linked to at the end of this paragraph doesn't really define what "implicit provider configuration" would mean here... it talks about implicit _inheritance_ of _unaliased_ providers ... so it's the _passing_ that is implicit, rather than the configuration itself.

Rather than reworking all of our terminology on that other page to account for this weird case of a version constraint, perhaps it's clearer just to add another sentence to the end of this paragraph:

> If a `provider` block is present in a child module then it overrides the implicit inheritance behavior, so either a full configuration must be provided in that block or a configuration must be passed explicitly from the parent module.

Something similar should probably be added to the module usage page too, but the upgrade guide is the priority for the moment, while people are working on upgrading, and we can take our time to figure out how best to describe this exception in the other page once the dust has settled and it's clear what the new pattern is.

Reply:
Ah you're right, that wording isn't consistent now that I reread it. I was going to change it to `implicitly used providers`, but I like your sentence better :+1: